File: solutions/csharp/difference-of-squares/1/DifferenceOfSquares.cs
using System;
using System.Linq;

public static class DifferenceOfSquares
{
    public static int CalculateSquareOfSum(int max)
    {       
        var sum = Enumerable.Range(1, max).Sum();
        return sum * sum;
    }

    public static int CalculateSumOfSquares(int max)
    {
        return Enumerable.Range(1, max).Sum(n => n * n);
    }

    public static int CalculateDifferenceOfSquares(int max)
    {
        return (CalculateSquareOfSum(max)) - (CalculateSumOfSquares(max));
    }
}

File: solutions/fsharp/difference-of-squares/1/DifferenceOfSquares.fs
module DifferenceOfSquares

let squareOfSum number = 
    [ 1 .. number ] |> List.sum |> (function n -> n * n)

let sumOfSquares number =
    [ 1 .. number ] |> List.sumBy(fun n -> n * n)

let differenceOfSquares number = 
    (squareOfSum number) - (sumOfSquares number)